src/concretize.rs

```diff
@@ -0,0 +1,282 @@
+//! Generic [`PUT`] trait defining an interface with a TLS library with which we can:
+//! - [`new`] create a client (or server) new initial state + bind buffers
+//! - [`progress`] makes a state progress (interacting with the buffers)
+//!
+//! And specific implementations of PUT for the different PUTs.
+use crate::agent::{AgentName, TLSVersion};
+use crate::error::Error;
+use crate::io::MessageResult;
+use crate::io::{MemoryStream, Stream};
+use crate::openssl_binding::openssl_version;
+use crate::trace::VecClaimer;
+use crate::wolfssl_binding;
```

Comment:
Can be done later: I think we should make it possible to enable/disable openssl or wolfssl via feature flags.

* Compiling wolfssl & openssl takes a long time to compile
* By depending expicitely on wolfssl and openssl we make a "closed set" of supported PUTs. At some point we probably will have to make `tlspuffin` a library instead of a binary. Then library consumers can add support for any protocol to `tlspuffin`. That will probably be a task for my internship.